### PR TITLE
Use Poetry for dependency management and add support for AWS CloudShell

### PR DESCRIPTION
**Pull request by liz-acosta**
This PR switches to Poetry for dependency management and uses Poetry to add support for AWS CloudShell via:

- Bash script that installs Poetry and Terraform
- Make target that calls this script

The PR updates the README to reflect these changes.

This PR also:

- Adds mugc.py as a single file instead of pulling in whole editable repo
- Changes font color for better visibility in dark mode

Tested it all in CloudShell and it works 😸

### CODE REVIEW

**Review by ajkerrigan — APPROVED** [review]
Looks good, thank you!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+description = "CExample policies, infrastructure, and helpful automation for learning c7n"
+authors = ["Cloud Custodian Project"]
+readme = "README.md"
+homepage = "https://cloudcustodian.io"
+repository = "https://github.com/cloud-custodian/examples"
+
+[tool.poetry.dependencies]
+python = "^3.7"
+c7n = "^0.9.14"
+appdirs = "1.4.4"
```
Comment (ajkerrigan) [inline pyproject.toml:13]: We can probably remove some/all of the deps listed after `c7n` here in a follow-up change. Poetry will keep track of pinned versions of lower level dependencies in its lock file so we don't have to pin as strictly or list all the individual side packages here.

Comment (liz-acosta) [reply pyproject.toml:13]: Cool. I'll file a ticket for that!

Comment (liz-acosta) [reply pyproject.toml:13]: Thank yooou!!!

**Pull request merged**